packages/js/components/src/date-time-picker-control/date-time-picker-control.tsx

```diff
@@ -174,72 +174,118 @@ export const DateTimePickerControl: React.FC< DateTimePickerControlProps > = ( {
 	// - [useDebounce hook loses function calls if the dependency changes](https://github.com/WordPress/gutenberg/issues/35505)
 	// - [useMemoOne and useCallbackOne](https://github.com/alexreardon/use-memo-one)
 
-	const onChangePropFunctionRef = useRef<
+	const onChangeRef = useRef<
 		DateTimePickerControlOnChangeHandler | undefined
 	>();
 	useLayoutEffect( () => {
-		onChangePropFunctionRef.current = onChange;
+		onChangeRef.current = onChange;
 	}, [ onChange ] );
 
-	function inputStringChangeHandlerFunction(
-		newInputString: string,
-		fireOnChange: boolean
-	) {
-		if ( ! isMounted.current ) return;
+	function setInputStringWithMoment( dateTime: Moment ) {
```

Comment:
What do you think about renaming this to `formatAndSetInputString`?  Or better yet, just `formatDateTime ` and call `setInputString` afterwards.

I'll leave another comment below in case that's not clear.

Reply:
💯 Much, much better.